components/service-binding-usage-controller/internal/controller/controller.go

```diff
@@ -127,13 +136,18 @@ func NewServiceBindingUsage(
 		kindsSupervisors:         kindSupervisors,
 		podPresetModifier:        podPresetModifier,
 		labelsFetcher:            labelsFetcher,
+		guard:                    sbuGuard,
 		maxRetires:               defaultMaxRetries,
 		log:                      log.WithField("service", "controller:service-binding-usage"),
 		queue:                    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ServiceBindingUsage"),
 		prefixGetter:             &envPrefixGetter{},
 		metric:                   cbm,
 	}
 
+	bindingInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: c.triggerServiceBindingUsageReconciliation,
```

Comment:
what about addFunc?

Reply:
`addFunc` is not necessary, in the first iteration (add action) of controller binding, SB is initiating with empty status value (https://github.com/kubernetes-sigs/service-catalog/blob/master/pkg/controller/controller_binding.go#L181-L188)

Reply:
Imagine scenario:
1. Create SBU but SB is not existing.
2. The controller is being restarted.
3. SB is created and get ready (status ready)
4. The controller started (after few seconds)
4.1 The controller is getting Add event for ServiceBinding

After the start of the controller, Binding is not updated, so the controller won't be triggered

Reply:
Then IMO both events are triggered, add and update.
I tested and it works, what I did:
- create SBU without SB, after 15 attempts SBU failed
- scale down SBU controller
- create SB, wait it will be ready/true
- scale-up SBU controller
SBU status immediately changes to ready/true

Reply:
It seems the SBU controller got Add event and processed just after scaling the controller up.